examples/cndpfwd/l3-fwd.c

```diff
@@ -0,0 +1,80 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ * Copyright (c) 2022 Intel Corporation.
+ */
+
+#include <cne_fib.h>           // For fib_create, fib_add...
+#include <cne_inet4.h>         // for inet_mtoh64 and inet_h64tom
+#include <net/cne_ether.h>     // for ether_addr
+
+#include "main.h"
+
+#define IPV4_L3FWD_NUM_FIB_RULES \
+    (sizeof(ipv4_l3fwd_fib_rule_array) / sizeof(ipv4_l3fwd_fib_rule_array[0]))
+
+struct ipv4_l3fwd_fib_rule {
+    uint32_t ip;
+    uint8_t depth;
+    struct ether_addr nh;
+    uint16_t tx_port;
+};
+
+static struct cne_fib *fib;
+
+static int
+l3fwd_fib_populate(struct cne_fib *fib)
+{
+    /*
+     * 198.18.0.0/16 are set aside for RFC2544 benchmarking (RFC5735).
+     * Each FIB rule is IPv4, depth, next hop destination MAC, TX port.
+     */
+    struct ipv4_l3fwd_fib_rule ipv4_l3fwd_fib_rule_array[] = {
+        {CNE_IPV4(198, 18, 0, 0), 24, {{0x02, 0x00, 0x01, 0x02, 0x03, 0x04}}, 1},
+        {CNE_IPV4(198, 18, 1, 0), 24, {{0x06, 0x00, 0x01, 0x02, 0x03, 0x04}}, 0}
+    };
+
+    for (uint16_t i = 0; i < IPV4_L3FWD_NUM_FIB_RULES; i++){
+        uint32_t ip = ipv4_l3fwd_fib_rule_array[i].ip;
+        uint8_t depth = ipv4_l3fwd_fib_rule_array[i].depth;
+        uint64_t eaddr = 0;
+
+        inet_mtoh64(&ipv4_l3fwd_fib_rule_array[i].nh, &eaddr);
+
+        /* Store both TX port and destination MAC in FIB's nexthop field. */
+        uint64_t nexthop = ((uint64_t)ipv4_l3fwd_fib_rule_array[i].tx_port << 48) | eaddr;
+
+        if(cne_fib_add(fib, ip, depth, nexthop) < 0)
+            return -1;
+    }
+
+    return 0;
+}
+
+int
+l3fwd_fib_lookup(uint32_t *ip, struct ether_addr *eaddr, uint16_t *tx_port)
+{
+    uint64_t nexthop;
+
+    cne_fib_lookup_bulk(fib, ip, &nexthop, 1);
```

Comment:
While I'm here, also wanted to ask a maintainer's opinion on this call. `cne_fib_lookup_bulk` appears to be designed to handle arrays of `N` lookups, where in this case `N` is always 1. We call this function in the `for` loop over the array of received packets.

We didn't have time to look into this, but would it be more idiomatic if we were to set up those arrays in that RX loop, make a single call to `cne_fib_lookup_bulk`, then rewrite and TX the packets in a loop, as opposed to the current approach where we RX->lookup->rewrite->TX in a loop?

Reply:
I believe it would be a good change. It just means the arrays for the FIB lookup would need to be created dynamically using a dynamic array based on n_pkts or maximum statically created arrays. Using calloc or malloc here would be a performance hit.

Reply:
Will create an enhancement issue for making this change once this PR is merged.